### PR TITLE
Start time fix

### PR DESCRIPTION
**Pull request by David-Baddeley**
Addresses issue with event timing when there are lots of initial (-1) protocol tasks (should also generally improve robustness of timing).

Also includes a few event handling and display fixes.

### CODE REVIEW

**Comment by David-Baddeley** [comment]
@barentine - you might be interested in this

File: PYME/Acquire/Spooler.py [hunk 1]
```diff
@@ -133,14 +133,14 @@ def StartSpool(self):
 
         self.imNum = 0
 
-        # record start time here in case protocol init tasks generate events (prob only effects simulator).
-        self.tStart = time.time()
-
         self.protocol.Init(self)
 
         self._collect_start_metadata()
-
```
Comment (barentine) [inline PYME/Acquire/Spooler.py:142]: This probably wont work quite right. I think we need to set `self.tStart` before `_collect_Start_metadata` where we log it

Comment (David-Baddeley) [reply PYME/Acquire/Spooler.py:142]: good catch ...

**Comment by David-Baddeley** [comment]
@barentine - have you actually run acquisition with this already? Or should I check on the simulator prior to merging ...

**Comment by barentine** [comment]
I was using it, yes. Am now concerned that you found other things to fix in a87af8e

**Comment by barentine** [comment]
though if I understand at quick glance that appears to be a) simulator specific, b) display related, and c) only an issue for spooling to h5 files. Correct?

**Comment by David-Baddeley** [comment]
Yep, 3 separate small regressions, one simulator specific. An event display regression in dh5view, and a PYMEClusterOfOne regression which I think is actually separate to the changes in this PR (think it was #922).

**Pull request merged**